src/Resources/config/core.xml

```diff
@@ -18,22 +18,31 @@
             <argument type="service" id="service_container"/>
             <tag name="routing.loader"/>
         </service>
+        <service id="Sonata\AdminBundle\Route\AdminPoolLoader" alias="sonata.admin.route_loader"/>
         <service id="sonata.admin.helper" class="Sonata\AdminBundle\Admin\AdminHelper" public="true">
             <argument type="service" id="sonata.admin.pool"/>
         </service>
+        <service id="Sonata\AdminBundle\Admin\AdminHelper" alias="sonata.admin.helper"/>
         <service id="sonata.admin.builder.filter.factory" class="Sonata\AdminBundle\Filter\FilterFactory" public="true">
             <argument type="service" id="service_container"/>
             <argument/>
         </service>
+        <service id="Sonata\AdminBundle\Filter\FilterFactory" alias="sonata.admin.builder.filter.factory"/>
         <service id="sonata.admin.breadcrumbs_builder" class="Sonata\AdminBundle\Admin\BreadcrumbsBuilder" public="true">
             <argument>%sonata.admin.configuration.breadcrumbs%</argument>
         </service>
+        <service id="Sonata\AdminBundle\Admin\BreadcrumbsBuilder" alias="sonata.admin.breadcrumbs_builder"/>
         <!-- Services used to format the label, default is sonata.admin.label.strategy.noop -->
         <service id="sonata.admin.label.strategy.bc" class="Sonata\AdminBundle\Translator\BCLabelTranslatorStrategy" public="true"/>
+        <service id="Sonata\AdminBundle\Translator\BCLabelTranslatorStrategy" alias="sonata.admin.label.strategy.bc"/>
```

Comment:
Unsure about this one: people are not supposed to type hint against concrete classes like this one, but against interfaces, so thiis alias shouldn't be useful for autowiring, should it?

Reply:
If someone typehint with the interface and want to inject a specific class, he may prefer adding in the config this line
```
Sonata\AdminBundle\Translator\LabelTranslatorStrategyInterface: '@Sonata\AdminBundle\Translator\BCLabelTranslatorStrategy'
```
over this one
```
Sonata\AdminBundle\Translator\LabelTranslatorStrategyInterface: '@sonata.admin.label.strategy.bc'
```

Reply:
Why we should need to create an alias for `Sonata\AdminBundle\Translator\BCLabelTranslatorStrategy`?
IMO, the natural autowiring use case for this is the one @greg0ire mentioned:
```php
use Sonata\AdminBundle\Translator\LabelTranslatorStrategyInterface;
// ...
public function __construct(LabelTranslatorStrategyInterface $labelTranslator)
{
    // ...
}
```
If my requirement is based on a specific implementation other than the default one provided by the bundle, the responsibility of defining a new service or overriding some existing definition must be up to me.

Reply:
The syntax `@sonata.admin.label.strategy.bc` is deprecated by symfony, and the classname should be prefer as id.
You can see it's already done in some others config (example: https://github.com/sonata-project/SonataAdminBundle/blob/3.x/src/Resources/config/commands.xml).

I think Sonata should migrate all legacy id, maybe in the next major. Creating alias is the best way to start the migration.

In your example, the user has to write some config to explain which class use for the interface.
Actually, the user has to write: 
```
Sonata\AdminBundle\Translator\LabelTranslatorStrategyInterface: '@sonata.admin.label.strategy.bc'
```
After my PR, he can write instead
```
Sonata\AdminBundle\Translator\LabelTranslatorStrategyInterface: '@Sonata\AdminBundle\Translator\BCLabelTranslatorStrategy'
```
Which is better, since it use the preferred syntax from Symfony.

Reply:
>The syntax @sonata.admin.label.strategy.bc is deprecated by symfony

I think the proposed changes in this PR are based on this misconception. Using an arbitrary non classname service name is not deprecated. For end user projects, the recommended way is to rely on [autoconfigured service ids](https://symfony.com/doc/current/service_container.html#service-container-services-load-example), which use the classnames for that purpose. Even, the docs encourages to leverage this feature, for end user projects; but that doesn't mean the non autowirable definitions are deprecated.
By other hand, reusable bundles (like this) [should not rely on autowiring internally](https://symfony.com/doc/current/service_container/autowiring.html#public-and-reusable-bundles). In order to promote a better UX, they should expose autowirable services for the classes that are part of their public API in order to let end users to ease the injection of those services.

>In your example, the user has to write some config to explain which class use for the interface.
>Actually, the user has to write:
>
>```
>Sonata\AdminBundle\Translator\LabelTranslatorStrategyInterface: '@sonata.admin.label.strategy.bc'
>```

That should not be required if who is using the autowirable service just need the same service which is proposed as the default implementation by this bundle. By instance:
```xml
<service id="Sonata\AdminBundle\Translator\LabelTranslatorStrategyInterface" alias="sonata.admin.label.strategy.native"/>
```

Finally, if the users require a different implementation that the proposed by the bundle, is up to them to [configure the service that they need](https://symfony.com/doc/current/service_container/autowiring.html#dealing-with-multiple-implementations-of-the-same-type) (overriding the service like in your example or wiring it manually).

For more details, take a look at https://symfony.com/doc/current/service_container/autowiring.html#working-with-interfaces. Even, you can just run `bin bin/console debug:container` in your project and you'll see how the services in your project are defined.

Reply:
My use case was the following. Since I found no way to config all the admin (and I have to set the label translator strategy and the model manager in every admin config), I created a `BaseAbstractAdmin` with this function.
```
/**
     * @required
     *
     * @param UnderscoreLabelTranslatorStrategy $labelTranslatorStrategy
     * @param ModelManager                      $modelManager
     */
    public function setDefaultConfig(
        UnderscoreLabelTranslatorStrategy $labelTranslatorStrategy,
        ModelManager $modelManager
    ) {
        $this->setLabelTranslatorStrategy($labelTranslatorStrategy);
        $this->setModelManager($modelManager);
    }
```

Since I want specific labelTranslatorStrategy and modelManager, I thought it was kinda weird to still using interface and then writing somewhere in my config which service should be injected.

So your point is to remove alias for service with interfaces ?

Reply:
If you need to replace the services this bundle provides by default, you should take a look at https://sonata-project.org/bundles/admin/3-x/doc/reference/configuration.html or give a try to the `config:dump-reference sonata_admin` command. Otherwise you should call the corresponding setter for each admin service.
BTW, I didn't say that an alias must be removed.

Reply:
@phansys 
Sure, I would like to set with the config
```
admin_services:
        model_manager: null
        label_translator_strategy: null
```
But it doesn't work this way https://github.com/sonata-project/SonataAdminBundle/issues/3190 and has to be admin by admin.

So I call the setter in the admin service, in a parent class I created. One unique call.
But that's another story.

For this PR, if you didn't say that an alias must be removed, is it ok for you to remove the `requested change` flag ?

Reply:
This is turning confusing, let me clarify: I'm requesting that this PR don't should create aliases for classes that are part of the implementation. I'm fine with:
* defining an alias with `Sonata\AdminBundle\Translator\LabelTranslatorStrategyInterface` as service id, pointing to the default label translator strategy;
* defining aliases with `Sonata\AdminBundle\Translator\*LabelTranslatorStrategy` as service ids, pointing to the defined implementations.

BTW, given the changes you are proposing, I think you should take into account this compiler pass:
https://github.com/sonata-project/SonataAdminBundle/blob/64b83e290566dc609a7a30d6a419c1359430dfc9/src/DependencyInjection/Compiler/AddDependencyCallsCompilerPass.php#L281

Regarding the configuration nodes `admin_services.model_manager` and `label_translator_strategy`, if they are not working as supposed to be, I think we should try to fix them. Please, can you confirm this?

Reply:
I think I understood, and I added
<service id="Sonata\AdminBundle\Translator\LabelTranslatorStrategyInterface" alias"sonata.admin.label.strategy.native"/>

Reply:
Great! :+1: 
Could you please add tests covering the availability of these new definitions?

Reply:
Sure, I can take time to write tests but... I never did test about something like this. And I didn't find it in this project. Can you elaborate your suggestion ?

You mean creating something like tests/Ressources/config/CoreTest.php in which I call the container with the new Id ?

Reply:
I think you could add the test for these services at `SonataAdminExtensionTest`:
https://github.com/sonata-project/SonataAdminBundle/blob/64b83e290566dc609a7a30d6a419c1359430dfc9/tests/DependencyInjection/SonataAdminExtensionTest.php#L20

Reply:
Thanks, I added them